src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -20,81 +28,90 @@ public enum armPosition {
+    armPIDController.setFF(Constants.ArmConstants.ARM_FF);
+    armPIDController.setOutputRange(
+        Constants.ArmConstants.ARM_MIN_OUTPUT, Constants.ArmConstants.ARM_MAX_OUTPUT);
+
+    leftMotor.setIdleMode(Constants.ArmConstants.kArmMotorIdleMode);
+    rightMotor.setIdleMode(Constants.ArmConstants.kArmMotorIdleMode);
+    leftMotor.setSmartCurrentLimit(Constants.ArmConstants.kArmMotorCurrentLimit);
+    rightMotor.setSmartCurrentLimit(Constants.ArmConstants.kArmMotorCurrentLimit);
+    leftMotor.setInverted(Constants.ArmConstants.RIGHT_MOTOR_INVERTED);
 
     // setting soft limits (soft limits keep the motor running when it hits the limit instead of
     // braking)
-    this.rightMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kForward, true);
-    this.rightMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, FORWARD_LIMIT);
-    this.rightMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, true);
-    this.rightMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, REVERSE_LIMIT);
+    leftMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kForward, true);
+    leftMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, Constants.ArmConstants.FORWARD_LIMIT);
+    leftMotor.enableSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, true);
+    leftMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse,  Constants.ArmConstants.REVERSE_LIMIT);
 
     // setting the idle mode setting for the SparkMax, here it is set to brake when idle
-    this.rightMotor.setIdleMode(CANSparkMax.IdleMode.kBrake);
-
-    // motor type for left motor
-    this.leftMotor = new CANSparkMax(leftMotorCANID, MotorType.kBrushless);
-
-    // comment below is for incase the motors need to be reversed
-    // this.leftMotor.setInverted(invertLeftMotor);
+    rightMotor.follow(this.leftMotor, Constants.ArmConstants.LEFT_MOTOR_INVERTED);
 
-    this.leftMotor.follow(this.rightMotor, true);
+    leftMotor.burnFlash();
+    rightMotor.burnFlash();
 
-    armPositions.put(armPosition.INTAKE_ARM_POSITION_GROUND, INTAKE_ARM_GROUND);
-    armPositions.put(armPosition.INTAKE_ARM_POSITION_SHELF, INTAKE_ARM_SHELF);
-    armPositions.put(armPosition.SCORING_ARM_POSITION_LOW, SCORING_ARM_LOW);
-    armPositions.put(armPosition.SCORING_ARM_POSITION_MID, SCORING_ARM_MID);
+    armPositions.put(armPosition.INTAKE_ARM_POSITION_GROUND,  Constants.ArmConstants.INTAKE_ARM_GROUND);
+    armPositions.put(armPosition.INTAKE_ARM_POSITION_SHELF,  Constants.ArmConstants.INTAKE_ARM_SHELF);
+    armPositions.put(armPosition.SCORING_ARM_POSITION_LOW,  Constants.ArmConstants.SCORING_ARM_LOW);
+    armPositions.put(armPosition.SCORING_ARM_POSITION_MID, Constants.ArmConstants. SCORING_ARM_MID);
   }
 
   // setting functions of runToScore and runToIntake
   public void runToScoreLow() {
-    this.rightMotor.set(FORWARD_SPEED);
+    this.rightMotor.set( Constants.ArmConstants.FORWARD_SPEED);
     this.currentArmPosition = armPosition.SCORING_ARM_POSITION_LOW;
```

Comment:
Line 93 (`this.rightMotor.set(speed)`) will cause problems: this will directly set the arm motors to spin with a constant output. The intent here was definitely to set the speed that it moves at while going to the position, which makes sense.

I initially thought that line 94 didn't actually do anything, but I see that it is handled by moveArmToPosition later down. A comment to that affect or adding in a call to moveArmToPosition would clarify the code and make it easier to use.

Reply:
Once Egan has marked this as resolved I think I can move forward with the merge.